localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -819,67 +822,91 @@ std::array<double, 36> NDTScanMatcher::estimate_covariance(
     std::stringstream message;
     message << "Error in Eigen solver: " << e.what();
     RCLCPP_WARN_STREAM_THROTTLE(this->get_logger(), *this->get_clock(), 1000, message.str());
-
-    return param_.covariance.output_pose_covariance;
   }
 
-  // first result is added to mean
-  const int n =
-    static_cast<int>(param_.covariance.covariance_estimation.initial_pose_offset_model.size()) + 1;
-  const Eigen::Vector2d ndt_pose_2d(ndt_result.pose(0, 3), ndt_result.pose(1, 3));
-  Eigen::Vector2d mean = ndt_pose_2d;
-  std::vector<Eigen::Vector2d> ndt_pose_2d_vec;
-  ndt_pose_2d_vec.reserve(n);
-  ndt_pose_2d_vec.emplace_back(ndt_pose_2d);
-
   geometry_msgs::msg::PoseArray multi_ndt_result_msg;
   geometry_msgs::msg::PoseArray multi_initial_pose_msg;
   multi_ndt_result_msg.header.stamp = sensor_ros_time;
-  multi_ndt_result_msg.header.frame_id = param_.frame.map_frame;
   multi_initial_pose_msg.header.stamp = sensor_ros_time;
+  multi_ndt_result_msg.header.frame_id = param_.frame.map_frame;
   multi_initial_pose_msg.header.frame_id = param_.frame.map_frame;
   multi_ndt_result_msg.poses.push_back(matrix4f_to_pose(ndt_result.pose));
   multi_initial_pose_msg.poses.push_back(matrix4f_to_pose(initial_pose_matrix));
 
```

Comment:
Rather than receiving the base covariance as an argument, how about declare it as a local variable on this line?
```
  std::array<double, 36> ndt_covariance = param_.covariance.output_pose_covariance;
```

Reply:
This is one option.
However, it requires using ndt_covariance converted to the base_link coordinate system.
Therefore, the concern is that the code will become longer and calculations will increase.